6_nodejs/week3/app.js

```diff
@@ -0,0 +1,49 @@
+import knex from 'knex'
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3307,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "password",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+//Preventing SQL Injection
+contactsAPIRouter.get("/", async (req, res) => {
+    try{
+        const { sort } = req.query;
+        const allowedSortKeys = ["id", "first_name", "last_name", "email", "phone"];
```

Comment:
You solved the issue of SQL injection vulnerability via [allowlisting](https://planetscale.com/blog/how-to-prevent-sql-injection-attacks-in-node-js#allowlisting). Great! 👍 One thing: the original code allowed the user to determine the sorting direction: either `ASC` or `DESC`. You took away this feature, but you can easily re-introduce it via an extra query parameter, which you should also allowlist.